pkg/vm/engine/tae/containers/cn_vectorwindow_adapter.go

```diff
@@ -0,0 +1,190 @@
+// Copyright 2022 Matrix Origin
```

Comment:
How much can we remove?

Reply:
Right now, the window is forwarding most of the calls to the underlying `cn_vector_adapter`. I don't think, we can remove a lot from this class as most of the functions in this class depend on the `window.offset`. Introducing window.offset in the cn_vector_adapter would make it cluttered.